examples/demo-nft-module/README.md

```diff
@@ -0,0 +1,481 @@
+
+Let's start with defining basic dependencies in `Cargo.toml`:
+
+```toml
+[dependencies]
+anyhow = { anyhow = "1.0.62" }
+sov-modules-api = { git = "https://github.com/Sovereign-Labs/sovereign.git", branch = "main", default-features = false }
+sov-modules-macros = { git = "https://github.com/Sovereign-Labs/sovereign.git", branch = "main" }
+```
+
+## Module definition
+
+NFT module is defined as following:
+
+```rust
+use sov_modules_api::Context;
+use sov_modules_macros::ModuleInfo;
+
+#[derive(ModuleInfo, Clone)]
+pub struct NonFungibleToken<C: Context> {
+    #[address]
+    pub address: C::Address,
+
+    #[state]
+    pub(crate) admin: sov_state::StateValue<C::Address>,
+
+    #[state]
+    pub(crate) owners: sov_state::StateMap<u64, C::Address>,
+}
+```
+
+Where it has:
+
+1. Address. Each module has to have an address. A module address is similar to a smart contract address in Ethereum. For example, a module can receive tokens. The address is calculated from a hash of the module prefix. This means that:
+    - The module address is unique
+    - No one knows the private key that produces this address
+
+2. Several state attributes. In this case it is an address of admin, and map of token id to owner address. Token id is u64 for simplicity.
+
+
+Context is a trait which allows the runtime to pass verified data to modules during execution. 
+Currently the only required method in Context sender(), which returns the address of the person who initiated the transaction (the signer).
+Context also inherits the Spec trait, which defines the concrete types used by the rollup for Hashing, persistent data Storage, digital Signatures, and Addresses. 
+The Spec trait allows rollups to easily tailor themselves to different ZK VMs. 
+By being generic over a Spec, a rollup can ensure that any potentially SNARK-unfriendly cryptography can be easily swapped out.
+
+
+# Implementing `sov_modules_api::Module` trait
+Before implementing `Module` trait, several things needs to be done:
+
+1. Add new dependencies: serde, borsh, sov-state
+2. Define 'native' feature flag to separate logic that is not needed in ZK context
+3. Define Call and Query messages
+4. Define Config
+
+## Preparation
+
+
+
+1. Define `native` feature in `Cargo.toml`:
+
+```toml
+[dependencies]
+anyhow = "1.0.62"
+borsh = { version = "0.10.3", features = ["bytes"] }
+serde = { version = "1", features = ["derive"] }
+serde_json = "1"
+
+sov-modules-api = { git = "https://github.com/Sovereign-Labs/sovereign.git", branch = "main", default-features = false }
+sov-modules-macros = { git = "https://github.com/Sovereign-Labs/sovereign.git", branch = "main" }
+sov-state = { git = "https://github.com/Sovereign-Labs/sovereign.git", branch = "main", default-features = false }
+
+
+[features]
+default = ["native"]
+serde = ["dep:serde", "dep:serde_json"]
+native = ["serde", "sov-state/native", "sov-modules-api/native"]
```

Comment:
When reading I was a bit unclear how "defining" the native flag look like. 

After the "Define 'native' feature flag to separate logic that is not needed in ZK context" would it make sense to explain why in this instance the native flag definition includes "serde", and "sov-state/native" etc.?

Reply:
Addressed this